### ISSUE TITLE
Combobox is missing aria-controls

### ISSUE THREAD

**Issue by kilgharsap** [SAP/ui5-webcomponents#3840]
### **Bug Description**

Running A11y testing, finding out Combobox is missing aria-controls, running the axe chrome-extension reporting aria-controls.

### **Expected Behavior**
A clear and concise description of what you expect to happen.

Running A11y testing, finding out Combobox is missing aria-controls, running the axe chrome-extension reporting aria-controls.

### **Steps to Reproduce**
1. Go to https://sap.github.io/ui5-webcomponents/master/playground/components/ComboBox/
2. Opens Inspector on Google chrome browser, and run 'axe dev' scan, reporting aria-controls is missing 
3. ...

### **Isolated Example**
If possible, please provide an isolated example by forking [CodeSandbox](https://codesandbox.io/s/71r1x5o51q?fontsize=14&module=%2Findex.html).

### **Context**
- UI5 Web Components version: 0.31.14
- OS/Platform: mac OS Big Sur
- Browser: Chrome Version 93.0.4577.63 (Official Build) (x86_64)
- Affected component: ComboBox

### **Log Output / Stack Trace / Screenshots**
```
{...}
```
### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

The priority indicates the severity of the issue. To set the appropriate priority consider the following criteria:
- Breaks entire application or system - High or Very High
- Accessibility issue - Medium or High
- Functional issue - Medium or High
- Visual issue - Low or Medium

**Note:** The priority might be re-evaluated by the issue processor.

### **Stakeholder Info** (if applicable)

- Organization: {...}
- Bussiness impact: {...}
<img width="1157" alt="Screen Shot 2021-09-06 at 5 29 50 PM" src="https://user-images.githubusercontent.com/89424021/132268074-4045ebf9-4933-44a7-8e64-39af9c8de1b8.png">

**Comment by gmkv** [comment]
Hi @SAP/ui5-webcomponents-topic-rl 

There is an ACC issue reported by AXE about the ComboBox missing the `aria-controls` attribute.
Can you have a look?

PS. This is also reproducible with OpenUI5's ComboBox as well.

Kind regards,
Georgi

**Comment by niyap** [comment]
Hello @kilgharsap  ,

The issue is already reproted: #3827

We are aware of this issue and other issues, related to setting relationships between accessible elements via IDrefs. Setting IDRef relationships are currently not possible due to the nature of the custom elements and their shadow DOM. The issue is present not only for UI5 Web Components, but for every framework creating custom elements and using shadow DOM. This is a gap in the accessibility support and there is a working draft for a new AOM (Accessibility Object Model), which addresses this issue and many more and aim a better, JS based accessibility support for all web element.

You can check the working draft here https://github.com/WICG/aom and the w3c specification here https://wicg.github.io/aom/spec/ (again draft). What is more, you can check out there explainer and see that https://wicg.github.io/aom/demos/#12 Issue 2 references the use case you mentioned and is one of the main issues, which lead to creating the AOM. In future, when this becomes complete and has support for all main browsers, we will have the possibility to adapt the components and make sufficient enhancements in the accessibility support, but until them we do not have any possibility to set there kind of relationships.

What is more, aria-controls does not affect the speech output and there will be no change in the usage of the element with or without the attribute and screen reader users will not be affected.

Thanks for understanding!

Kind Regards,
Niya
